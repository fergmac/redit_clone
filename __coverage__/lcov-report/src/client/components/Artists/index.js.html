<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/components/Artists/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/client/components/Artists</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/34</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/7</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent, PropType<span class="cstat-no" title="statement not covered" >s } from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
import style from './style.css';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import Drawer from 'material-ui/Drawer';
import Genre from '../Genre';
import SiteIcon from 'material-ui/svg-icons/av/library-music';
// import data from '../../data.js';
import { connect } from 'react-redux';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span>
// was lessons
class Artists <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >ex</span>tends PureComponent {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span>
  render() {
    re<span class="cstat-no" title="statement not covered" >turn (</span>
      &lt;Drawer&gt;
        &lt;AppBar
          className={style.albumCollection}
          title={&lt;span&gt;HEARDit&lt;/span&gt;}
          iconElementLeft={
            &lt;IconButton&gt;
              &lt;SiteIcon /&gt;
            &lt;/IconButton&gt;
            }
        /&gt;
          {this.props.genres.map<span class="fstat-no" title="function not covered" >((</span>genre, index) =&gt; (&lt;Genre genre={genre} key={index} /&gt;))}
      &lt;/Drawer&gt;
    );
  }<span class="cstat-no" title="statement not covered" ></span>
}
&nbsp;
<span class="cstat-no" title="statement not covered" >Artists.propTypes = {</span>
  genres: PropTypes.array.isRequired,
  artists: PropTypes.array.isRequired,
};
const mapStateToProps <span class="cstat-no" title="statement not covered" >= state =<span class="fstat-no" title="function not covered" >&gt; ({</span></span>
  genres: state.genres,
  artists: state.artists,
});
&nbsp;
export default connect(mapStateToProps)(Artists);
&nbsp;
// const Categories = () =&gt; (
//   &lt;Drawer&gt;
//     &lt;AppBar
//       className={style.categories}
//       title={&lt;span&gt;REDit&lt;/span&gt;}
//       iconElementLeft={
//         &lt;IconButton&gt;
//           &lt;SiteIcon /&gt;
//         &lt;/IconButton&gt;
//      }
//     /&gt;
//     {data.weeks.map((week, index) =&gt; (&lt;Week week={week} key={index} /&gt;))}
//   &lt;/Drawer&gt;
// );
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Nov 09 2016 16:25:23 GMT-0800 (PST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
