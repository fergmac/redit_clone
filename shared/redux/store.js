import { createStore } from 'redux';
import reducers from './reducers';

const store = createStore(reducers, {});

console.log(store);

export default store;
