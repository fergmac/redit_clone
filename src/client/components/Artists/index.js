import React, { PureComponent, PropTypes } from 'react';
// import style from './style.css';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import Drawer from 'material-ui/Drawer';
import Genre from '../Genre';
import SiteIcon from 'material-ui/svg-icons/av/library-music';
// import data from '../../data.js';
import { connect } from 'react-redux';

// const styles = {
//   albumCollection: {
//     width: '100px',
//     left: '0',
//     backgroundColor: 'darkturquoise !important',
//   },
// };
// was lessons
class Artists extends PureComponent {
  render() {
    return (
      <Drawer>
        <AppBar
          // style={styles.albumCollection}
          title={<span>HEARDit</span>}
          iconElementLeft={
            <IconButton>
              <SiteIcon />
            </IconButton>
            }
        />
          {this.props.genres.map((genre, index) => (<Genre genre={genre} key={index} />))}
      </Drawer>
    );
  }
}

Artists.propTypes = {
  genres: PropTypes.array.isRequired,
  artists: PropTypes.array.isRequired,
};
const mapStateToProps = state => ({
  genres: state.genres,
  artists: state.artists,
});

export default connect(mapStateToProps)(Artists);

// const Categories = () => (
//   <Drawer>
//     <AppBar
//       className={style.categories}
//       title={<span>REDit</span>}
//       iconElementLeft={
//         <IconButton>
//           <SiteIcon />
//         </IconButton>
//      }
//     />
//     {data.weeks.map((week, index) => (<Week week={week} key={index} />))}
//   </Drawer>
// );
