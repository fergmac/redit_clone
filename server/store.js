import { createStore } from 'redux';
import reducers from '../src/shared/redux/reducers';

export default createStore(reducers, {});


