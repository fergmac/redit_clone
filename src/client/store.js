import { createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import reducers from '../shared/redux/reducers';
import remoteMiddleware from './socket/remoteMiddleware';
import socket from './socket';

const logger = createLogger();

const middlewares = applyMiddleware(reduxThunk, remoteMiddleware(socket), logger);

const store = createStore(reducers, middlewares);

export default store;
