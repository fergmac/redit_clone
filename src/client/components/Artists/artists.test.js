/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Artists from './index';
import { Provider } from 'react-redux';
import store from '../../store';


describe('Artists', () => {
    const artistsData = {
        id: 1,
        name: 'Eddie Lang',
    }
    it('renders the artists', () => {
       const wrapper = shallow(
           <Provider store={store}>
                <Artists 
                    artists={artistsData}
                />
          </Provider>
       )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})