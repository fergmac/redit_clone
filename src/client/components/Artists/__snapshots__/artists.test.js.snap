exports[`Artists renders the artists 1`] = `
<Connect(Artists)
  artists={
    Object {
      "id": 1,
      "name": "Eddie Lang",
    }
  } />
`;
